adapter/src/ethereum.rs

```diff
@@ -276,6 +266,49 @@ impl Adapter for EthereumAdapter {
     }
 }
 
+#[derive(Debug, Clone)]
+struct RelayerClient {
+    client: Client,
+    relayer_url: String,
+}
+
+impl RelayerClient {
+    pub fn new(relayer_url: &str) -> Result<Self, reqwest::Error> {
+        let client = Client::builder().build()?;
+
+        Ok(Self {
+            relayer_url: relayer_url.to_string(),
+            client,
+        })
+    }
+
+    pub async fn get_privilege(&self, identity: &[u8; 20]) -> Result<u8, AdapterError> {
+        use reqwest::Response;
+        use std::collections::HashMap;
+
+        let relay_url = format!(
+            "{}/identity/by-owner/{}",
+            self.relayer_url,
+            hex::encode(identity)
+        );
+
+        let identities_owned: HashMap<[u8; 20], u8> = self
+            .client
+            .get(&relay_url)
+            .send()
+            .and_then(|res: Response| res.json())
+            .await
+            .map_err(|_| map_error("Fetching privileges failed"))?;
+
+        let privilege = identities_owned
+            .get(identity)
+            .copied()
+            .unwrap_or_else(|| 0_u8);
```

Comment:
It shouldn't return `0_u8` if it fails to unwrap it, it means the identity isn't allowed. it should return an error `insufficient privilege`

Reply:
`0` is no privileges, and if you've seen the code that uses this function, you will see that we check if it's `> 0`

Reply:
We could make an enum, but not sure if you'll like that

Reply:
We don't need the privilege levels anymore, we can just return the error if we don't find it